src/app.js

```diff
@@ -126,36 +138,80 @@ export function app() {
 		 * @param {*=} data Data.
 		 * @returns {Promise}
 		 */
-		async call(method, target, data = {}) {
-			switch (method) {
-				case 'spawn':
-					const message = {
-						type: 'SPAWN',
-						target,
+		async spawn(target, data = {}) {
+			const message = {
+				type: 'SPAWN',
+				target,
+				data,
+				token
+			};
+
+			if (token) {
+				debug('%o to %o - %o', 'SPAWN', target, data);
+				postMessage(message);
+			} else {
+				debug('%o to %o - %o', 'SPAWN(queued)', target, data);
+				queueMessage(message);
+			}
+
+			// Wait for response from the app or some sort of failure
+			return new Promise(resolve => {
+				spawnResolve = resolve;
+			});
+		}
+	};
+
+	function handleSpawn({ data: message, source: sourceWindow }) {
+		debug('SPAWNED from %o - %O', message.source, message);
+
+		if (lifecycleHandlers.has('spawn')) {
+			/**
+			 * @TODO Timeout handling!
+			 */
+			lifecycleHandlers.get('spawn').apply({}, [
+				message.data,
+				function resolve(data) {
+					postMessage({
+						type: 'SPAWN-SUCCESS',
```

Comment:
do you think we should keep this name, or go with `SPAWN-SUBMIT` and `SPAWN-ERROR`?

Reply:
`TODO` for the future?

we could even separate the events from `apps` and to `apps`:

app -> hub - `SPAWN-SUBMIT` / `SPAWN-FAIL`
hub -> app - `SPAWN-SUBMITTED` / `SPAWN-FAILED` (or `SPAWNED-SUBMIT` / `SPAWNED-FAIL` )

------------

or go even crazier and just do:

app requesting spawn - `SPAWN` 
app submitting back - `SUBMIT`
app getting response from spawn - `type: 'SPAWNED', data: [err, data]`


of course this would be strange that while forwarding an event/message we change the `type`, so it might not be a good idea.